Modules/Module_Processing/xASL_init_BIDS2Legacy.m

```diff
@@ -95,7 +98,31 @@
 	x.dataPar = xASL_io_ReadJson(fListDataPar{1});
 end
 
+
 %% 4. Load the BIDS structure of all subjects
 x.modules.bids2legacy.BIDS = bids.layout(fullfile(x.dir.DatasetRoot,'rawdata'));
 
+
+%% 5. Create the derivatives directory
+if exist(x.dir.xASLDerivatives, 'dir')
+    fprintf('%s\n', [x.dir.xASLDerivatives ' already exists']);
+    fprintf('%s\n', 'Note that all pre-existing derivative subject folders will be overwritten,');
+    fprintf('%s\n', 'unless BIDS2Legacy lock files already exist for a subject');
+else
+    xASL_adm_CreateDir(x.dir.xASLDerivatives);
 end
+
+
+%% 6. Load BIDS configuration for file renaming
+x.modules.bids2legacy.bidsPar = xASL_bids_Config;
```

Comment:
OK. That is nice that you put bidsPar into a parameter within x. 
But then we have a lot of functions that use this bidsPar that get both bidsPar and x as an argument. So they will have doubled information.

The only solution is:
1. All functions that accept bidsPar as an input argument together with x will have to be rewritten.
2. The catch is that not all of these are inside bids2legacy, so we will have to check all calls of xASL_bids_Config and make sure that this is saved within the x-struct.

So that is a lot of extra work to be done. I can do that if you want but we have to be first careful that we don't leave this inconsistent code (cause we are cleaning, so we should clean properly) and we have to be careful that this is done correctly or it will leave us code that doesn't run...

Reply:
I see. This was mainly to make the code easier and avoid overlap between the import and processing modules. But more importantly, `bids_config` was now run at every xASL_module_BIDS2Legacy, and instead my edit runs it only once. This requires to put it in x (or to create many new bidsPar variables in `xASL_Iterate`. Just checked, and `xASL_module_BIDS2Legacy` provides this `x.modules.bids2legacy.bidsPar` instead of bidsPar to `xASL_bids_BIDS2Legacy_ParseModality`, which throughputs it to `xASL_bids_BIDS2Legacy_ParseScanType`. The latter two functions are unchanged. 

So the change for BIDS2Legacy's use of bidsPar is actually negligible. Are you OK with this?

Reply:
Yes - I wasn't afraid that it would have consequences. And it's a useful edit. But the question is - now that we have improved it on one place, we should probably do the improvement everyhwere. I can do that if you want...

Reply:
Sure, I'm fine with that. But that takes more effort. I would like to continue asap with cherry-picking my other stuff.

Reply:
OK. then can you create a new issue to finalize this later so that we don't have inconsistent code? We can do that later, but it would be good to do it in this release still.